var/spack/repos/builtin/packages/mxnet/package.py

```diff
@@ -128,4 +128,8 @@ def install(self, spec, prefix):
         # install python bindings
         if '+python' in spec:
             python = which('python')
-            python('python/setup.py', 'install', '--prefix={0}'.format(prefix))
+            # The python libs are in a separate dir, and it is necessary to change
+            # directory so that setup.py picks them up.
+            with working_dir('python'):
+                python('setup.py', 'install', '--prefix={0}'.format(prefix),
+                       '--single-version-externally-managed', '--root=/')
```

Comment:
```suggestion
                setup_py('install', '--prefix={0}'.format(prefix),
                         '--single-version-externally-managed', '--root=/')
```
This is set by the `python` package. It additionally adds `python setup.py --no-user-cfg`, which helps with reproducible builds. Then you can get rid of `which('python')`, which may not work for `python3`.

Reply:
Ok, I'll see if I can use the PythonPackage stuff by overloading the install function to be in the right directory.

Reply:
I didn't understand but now I do, there's no need to inherit from PythonPackage to get this setup_py function. I've pushed the update.